fastcrawler_ui/routers/spiders.py

```diff
@@ -49,32 +54,45 @@ async def clients(
     return items
 
 
-@spider_router.post("/stop_task", status_code=status.HTTP_204_NO_CONTENT)
-async def stop_task(
-    task: TaskInput = Body(),
+def manage_tasks(
+    task_names: TasksInput,
+    crawler: FastCrawler,
+    spider_repository: SpiderRepository,
+    spider_controller: Type[SpiderController],
+    action: str,
+):
+    controller = spider_controller(spider_repository)
+    action_call = getattr(controller, f"{action}_task_by_name")
+
+    tasks = [action_call(crawler, task_name) for task_name in task_names.names]
+
+    return asyncio.gather(*tasks)
```

Comment:
await asyncio gather :)

Reply:
This function is not `async` and written to return the `Future`.
The `await` process has done in the caller.

Reply:
then return type for the function should be a list of coroutine types
do the coroutine's typing completely so it's clear what coroutines would be returned

Reply:
it's better to return "task" and let the gather to be done in outer call, bcs function name suggests that it's returning tasks, not futures.